### DISCUSSION TITLE
How to do FEM electrical simulation on a given pore network?

### DISCUSSION THREAD

**Discussion by Camelot18** [PMEAL/porespy#727] [Q&A]
Dear Professor Jeff Gostick,

I need to conduct a FEM simulation on a pore network to obtain electrical conductivity, but it seems that you are still working on the conductance models algorithm. Do you have any suggestions to perform this?
Looking forward to your reply.

Best regards,
Peter

**Comment by jgostick** [comment]
Hi @Camelot18 Can you elaborate a bit?  I think your question spans both PoreSpy and OpenPNM right?

**Reply by Camelot18** [reply]
Dear Professor,

Thanks for your quick response.
Yes, the initial pore network was established via PoreSpy and OpenPNM. After adding hydrate to the pore network, there are three phases for the whole rock: brine in the pores and throats as a fluid phase(conductive), hydrate distributed in pore space as a solid phase(insulator), and grain particles as a solid phase(insulator). The conductivities for the three phases are known, I need to apply the rock a potential voltage and obtain the current density to calculate the total conductivity. I think it's similar to the permeability simulation, but no such algorithm is available for electrical properties.

Best,
Peter

**Reply by jgostick** [reply]
The fickian diffusion algorithm is analogous to electrical conduction.  FickianDiffusion is actually a subclass of GenericTransport (now just called Transport in V3), so you can use that.  You'll need to tell the algorithm what is the name of the quantity being solved for and where is the conductance stored.  You do this via the `settings` attribute.  You'll probably set `quantity='pore.voltage'` and `conductance='throat.electrical_conductance'`.  The trick will be compute the electrical conductance of each throat. There is a detailed example on this [here](https://openpnm.org/examples/reference/simulations/size_factors_and_transport_conductance.html)
